### PR TITLE
chore: add supervisor for lpt infra

### PR DESCRIPTION
**Pull request by NagyZoltanPeter**
# Description
This PR is based upon the previous one, and serves the same sole purpose: enable infra deployment.
Prev PR: https://github.com/waku-org/nwaku/pull/3158

# Changes

- After agreement with Marko: https://github.com/status-im/infra-misc/pull/380 we decided that infra deployment image must run continuously. As this is against how this test is designed it is now extended:
- Added lpt_supervisor.py script that takes default deployment arguments and a period time arg, and restarts tester apps in predefined times. (This has to be set carefully to allow both parties (SENDER/RECEIVER) to finish before next period comes.)
- infra.env : predefined deployment arguments to testing, built into the image
- if the image tag is set to deploy for docker build, it will create a specific image for infra deployment (having different entrypoint with lpt_supervisor.py)
- As a plus, as Franck requested, I added message latency metrics (latency was already reported but only to stdout).

### CODE REVIEW

**Review by gabrielmer — APPROVED** [review]
Thank you!

File: apps/liteprotocoltester/lpt_supervisor.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    for line in f:
+      if line.strip() and not line.startswith('#'):
+        key, value = line.strip().split('=', 1)
+        env_vars[key] = value
+  return env_vars
+
+def start_tester_node():
+  role = sys.argv[1]
+  env = os.environ.copy()
+  for key, value in env_vars.items():
```
Comment (markoburcul) [inline apps/liteprotocoltester/lpt_supervisor.py:20]: Where do you forward env_vars dictionary? I can't find it defined globally

File: apps/liteprotocoltester/lpt_supervisor.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  script_cmd = ['sh', '/usr/bin/run_tester_node_at_infra.sh', '/usr/bin/liteprotocoltester', role]
+  process = Popen(script_cmd, env=env)
+  return process
+
+if __name__ == "__main__":
+  if len(sys.argv) < 2 or sys.argv[1] not in ["RECEIVER", "SENDER"]:
+    print("Error: First argument must be either 'RECEIVER' or 'SENDER'")
+    sys.exit(1)
+
+  env_file = '/usr/bin/infra.env'
```
Comment (markoburcul) [inline apps/liteprotocoltester/lpt_supervisor.py:32]: Why are you hardcoding path to the env vars file? Wouldn't it be simpler to just check if the env var is defined and read it from the code?

File: apps/liteprotocoltester/Dockerfile.liteprotocoltester.compile [hunk 3]
```diff
@@ -43,15 +42,30 @@ EXPOSE 30303 60000 8545
...
 
 ENTRYPOINT ["/usr/bin/run_tester_node.sh", "/usr/bin/liteprotocoltester"]
 
-# # By default just show help if called without arguments
-CMD ["--help"]
+# Image for infra deployment -------------------------------------------
+FROM base_lpt AS deployment_lpt
+
+COPY --from=nim-build /app/apps/liteprotocoltester/run_tester_node_at_infra.sh /usr/bin/
+COPY --from=nim-build /app/apps/liteprotocoltester/infra.env /usr/bin/
```
Comment (markoburcul) [inline apps/liteprotocoltester/Dockerfile.liteprotocoltester.compile:66]: You can just define the env variables in this Dockerfile and set their default values. They can be changed if you pass them to the container when starting it.

File: Makefile [hunk 4]
```diff
@@ -373,16 +373,16 @@ DOCKER_LPT_NIMFLAGS ?= -d:chronicles_colors:none -d:insecure
...
 	docker build \
-	  --no-cache \
 		--build-arg="MAKE_TARGET=liteprotocoltester" \
 		--build-arg="NIMFLAGS=$(DOCKER_LPT_NIMFLAGS)" \
 		--build-arg="NIM_COMMIT=$(DOCKER_NIM_COMMIT)" \
 		--build-arg="LOG_LEVEL=TRACE" \
 		--label="commit=$(shell git rev-parse HEAD)" \
 		--label="version=$(GIT_VERSION)" \
-		--target $(TARGET) \
+		--target $(if $(filter deploy,$(DOCKER_LPT_TAG)),deployment_lpt,standalone_lpt) \
```
Comment (Ivansete-status) [inline Makefile:385]: what's the diff between `deployment_lpt` and `standalone_lpt`?

File: apps/liteprotocoltester/infra.env [hunk 5]
```diff
@@ -0,0 +1,11 @@
+TEST_INTERVAL_MINUTES=10
+START_PUBLISHING_AFTER=120
+NUM_MESSAGES=300
+DELAY_MESSAGES=1000
```
Comment (Ivansete-status) [inline apps/liteprotocoltester/infra.env:4]: Maybe we could have a more explicit var name;P
```suggestion
MSG_INTERVAL_MILLIS=1000
```

File: apps/liteprotocoltester/infra.env [hunk 6]
```diff
@@ -0,0 +1,11 @@
+TEST_INTERVAL_MINUTES=10
+START_PUBLISHING_AFTER=120
```
Comment (Ivansete-status) [inline apps/liteprotocoltester/infra.env:2]:
```suggestion
START_PUBLISHING_AFTER_SECS=120
```

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100: 
I just added some minor comments

File: Makefile (hunk 4 above)
Comment (NagyZoltanPeter) [reply Makefile:385]: It is the distinction between the different use. `deployment_lpt` stands for running in our infra environment with its own constraints. - main diff is that in deployment we need to run a supervisor ahead of tests that can control restarts.
`standalone_lpt` is the normal usage, so in lpt_runner you can easily run your tests on demand.

File: apps/liteprotocoltester/infra.env (hunk 5 above)
Comment (NagyZoltanPeter) [reply apps/liteprotocoltester/infra.env:4]: Good idea! I need to apply such change more deep. I will check!

File: apps/liteprotocoltester/infra.env (hunk 6 above)
Comment (NagyZoltanPeter) [reply apps/liteprotocoltester/infra.env:2]: Just as above, thanks for the suggestion.

File: Makefile (hunk 4 above)
Comment (Ivansete-status) [reply Makefile:385]: Thanks for clarifying! I was wondering if we could add a comment somewhere, if not done already:)

Comment (NagyZoltanPeter) [reply Makefile:385]: Yeah must add it to lpt readme at least!

**Review by markoburcul — APPROVED** [review]

**Pull request merged**